pcsx2-qt/Settings/DEV9SettingsWidget.cpp

```diff
@@ -493,7 +492,7 @@ void DEV9SettingsWidget::onEthHostDel()
 
 void DEV9SettingsWidget::onEthHostExport()
 {
-	std::vector<HostEntryUi> hosts = ListHostsConfig();
+	std::vector<HostEntryUi> hosts = ListHostsConfig().value();
```

Comment:
Is `ListsHostsConfig()` ever going to return nullopt here? If so, you should check that it has a value.

Reply:
The Export button is only accessible in the global settings panel, or the per-game settings panel when a per game host list is present, I `ListHostsConfig()` should never return nullopt in those cases.

I can however add a check regardless, if preferred

Reply:
Nah, it's fine, if it's never going to be nullopt there's no problem, was just concerned about it blowing up at runtime on some crazy config :)

Reply:
I did mean to clarify the logic, but I haven't done so until now

`ListHostsConfig()` only returns nullopt when it's call to `m_dialog->getIntValue("DEV9/Eth/Hosts", "Count", std::nullopt);` returns nullopt.
I can see that when getIntValue is not used with per game settings, std::nullopt gets treated as zero instead, so returning nullopt would only happen when `[DEV9/Eth/Hosts] Count` isn't found in the per game settings file*
In such a case, `RefreshHostList()` (called in the constructor) would disable the host Add/Delete/Export & Import buttons, which, as `onEthHostExport` is only called via a button click, effectively ensuring that code only running with valid settings

I’ll also add that `RefreshHostList()` assumes this behaviour and only checks for nullopt when working with per-game settings

In response to your comment, I’ve also briefly tested;
A clean Install, of which PCSX2 auto created the `Count` entry, which worked fine
Manually deleting the `Count` entry, which also worked fine, treating the missing entry as zero.

Reply:
I've since adjusted the logic so that `ListHostsConfig()` is no longer dependent on `m_dialog->getIntValue()` returning 0 for a default value of `std::nullopt` when not used with per game settings